### PR TITLE
[rllib] Minor cleanups to TFPolicyGraph: add init args, constants for loss inputs

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

- [x] Pass args explicitly to _init()
- [x] Use constants for loss inputs list instead of magic strings
- [x] Move postprocessing code out a bit into its own class

## Related issue number

https://github.com/ray-project/ray/issues/4418

### CODE REVIEW

**Comment by ericl** [comment]
jenkins retest this please

File: python/ray/rllib/agents/ppo/appo_policy_graph.py [hunk 1]
```diff
@@ -1,488 +1,497 @@
-"""Adapted from VTracePolicyGraph to use the PPO surrogate loss.
```
Comment (richardliaw) [inline python/ray/rllib/agents/ppo/appo_policy_graph.py]: wait what's the change in this file?

Comment (ericl) [reply python/ray/rllib/agents/ppo/appo_policy_graph.py]: Somehow the github diff is messed up, but it's similar to the vtrace_policy_graph change.

File: python/ray/rllib/agents/impala/vtrace_policy_graph.py [hunk 2]
```diff
@@ -94,12 +98,32 @@ def __init__(self,
         self.entropy = tf.reduce_sum(
             tf.boolean_mask(actions_entropy, valid_mask))
 
-        # The summed weighted loss
+        # The summpolicy logits, VF preds, anded weighted loss
```
Comment (richardliaw) [inline python/ray/rllib/agents/impala/vtrace_policy_graph.py]: typos

File: python/ray/rllib/agents/ppo/ppo.py [hunk 3]
```diff
@@ -71,30 +71,29 @@ class PPOAgent(Agent):
...
+                    "sgd_batch_size": config["sgd_minibatch_size"],
+                    "num_sgd_iter": config["num_sgd_iter"],
+                    "num_gpus": config["num_gpus"],
+                    "sample_batch_size": config["sample_batch_size"],
+                    "num_envs_per_worker": config["num_envs_per_worker"],
+                    "train_batch_size": config["train_batch_size"],
                     "standardize_fields": ["advantages"],
-                    "straggler_mitigation": (
-                        self.config["straggler_mitigation"]),
+                    "straggler_mitigation": (config["straggler_mitigation"]),
```
Comment (richardliaw) [inline python/ray/rllib/agents/ppo/ppo.py]:
```suggestion
                    "straggler_mitigation": config["straggler_mitigation"],
```

File: python/ray/rllib/agents/ddpg/ddpg_policy_graph.py [hunk 4]
```diff
@@ -11,23 +11,124 @@
...
+                 twin_q_t,
+                 twin_q_tp1,
+                 actor_loss_coeff=0.1,
+                 critic_loss_coeff=1.0,
+                 gamma=0.99,
+                 n_step=1,
+                 use_huber=False,
+                 huber_threshold=1.0,
+                 twin_q=False,
+                 policy_delay=1):
```
Comment (richardliaw) [inline python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: This could use a docstring

Comment (richardliaw) [reply python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: maybe in separate PR

**Review by richardliaw — APPROVED** [review]
I think the mixin Postprocessing is fine. One thing that would probably help is in `agents/README.md` or some dev-guide where you walk through (or show some flow chart) the structure of these agents, and also have docstrings for all of the components.

**Comment by ericl** [comment]
Sounds good... I'll make another PR with some ideas on simplifying agents and include that README there.
